pkg/controller/common/beat/metricbeat/metricbeat.go

```diff
@@ -0,0 +1,68 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package metricbeat
+
+import (
+	commonbeat "github.com/elastic/cloud-on-k8s/pkg/controller/common/beat"
+	"github.com/elastic/cloud-on-k8s/pkg/controller/common/container"
+	"github.com/elastic/cloud-on-k8s/pkg/controller/common/defaults"
+	"github.com/elastic/cloud-on-k8s/pkg/controller/common/reconciler"
+	"github.com/elastic/cloud-on-k8s/pkg/controller/common/volume"
+	corev1 "k8s.io/api/core/v1"
+)
+
+const (
+	Type commonbeat.Type = "metricbeat"
+
+	DockerSockVolumeName = "dockersock"
+	DockerSockPath       = "/var/run/docker.sock"
+	DockerSockMountPath  = "/var/run/docker.sock"
+
+	ProcVolumeName = "proc"
+	ProcPath       = "/proc"
+	ProcMountPath  = "/hostfs/proc"
+
+	CGroupVolumeName = "cgroup"
+	CGroupPath       = "/sys/fs/cgroup"
+	CGroupMountPath  = "/hostfs/sys/fs/cgroup"
+)
+
+type Driver struct {
+	commonbeat.DriverParams
+	commonbeat.Driver
+}
+
+func NewDriver(params commonbeat.DriverParams) commonbeat.Driver {
+	// use the default for metricbeat type if not provided
+	if params.DaemonSet == nil && params.Deployment == nil {
+		params.DaemonSet = &commonbeat.DaemonSetSpec{}
+	}
+
+	return &Driver{DriverParams: params}
+}
+
+func (d *Driver) Reconcile() (*commonbeat.DriverStatus, *reconciler.Results) {
+	f := func(builder *defaults.PodTemplateBuilder) {
+		dockerSockVolume := volume.NewHostVolume(DockerSockVolumeName, DockerSockPath, DockerSockMountPath, false, corev1.HostPathUnset)
+		procVolume := volume.NewReadOnlyHostVolume(ProcVolumeName, ProcPath, ProcMountPath)
+		cgroupVolume := volume.NewReadOnlyHostVolume(CGroupVolumeName, CGroupPath, CGroupMountPath)
```

Comment:
I think it's not possible currently to override the `podTemplate` so those volumes are not included. For example if you don't want to mount `/proc` because you don't have permissions to do so.
Let's not delay this PR for that reason though, I'll open another issue.